app/graphql/mutations/post/create_post.rb

```diff
@@ -56,6 +58,7 @@ def resolve(**kwargs)
       if post.save
         PostMailer.new_post(post)
         GoalReacher.check!(team)
+        send_post_announcement(post)
```

Comment:
This is a bit of a smell. See how this mutation needs to know about the postmailer, goalreacher and now also Slack? Ideally a new post would be an event, where all these things can hook into.

Put differently; what if there's suddenly another way of creating posts, not via GraphQL. Then that new way of creating posts would have to duplicate this code.

Reply:
Do you recommend hooking into the ActiveRecord create callback or creating a separate class that handles the creation of new posts and use that class for both GraphQL and Slack?

Reply:
This has been fixed by extracting the creation of post to a separate class.